analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -834,6 +867,10 @@ def main(args):
     if 'name' in args:
         metadata_tool['run_name'] = args.name
 
+    git_info = get_git_info(actions[0].directory)
```

Comment:
What if the build is an out of source build with a separate build directory Eg. if I compile xerces like this:
```sh
mkdir build
cd build
cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ../xerces-c
```
In this case the directory in the compile command json will be `"directory": "...tmp/xerces-c-build/src",`
which is not the xerces git repository.

The file would be probably a bit more precise, but I wonder what if multiple git repositories are included into one build (like the clang build before the monorepo). The detection of the git repository would depend on the first compile command in the json file, which depends in what order the build system calls the compilation.

Reply:
I am thinking on the following. Lets assume we have the following project structure:
```sh
home
 `- projects
   `- root_project_dir
     `- submodule1
       `- src1.cpp
      `- submodule2
       `- src2.cpp
```
If we log this the compilation database will contain 2 entries where the `file` key value will be the following:
- `/home/projects/root_project_dir/submodule1/src1.cpp`
- `/home/projects/root_project_dir/submodule2/src2.cpp`

In this case we can go up while the parent directory is a git repository. In our case we would find that the latest directory is `/home/projects/root_project_dir` which is our project root directory. This way we can get and save git information always on the project root directory.

What do you think @gyorb?

Reply:
Yes solution is better but I'm still a bit concerned that the detection depends only the compilation action order. If multiple non submodule or subtree git repositories are involved in a build it could detect another git repository if the compilation order changes.

Would be an extra `--repository` flag which points to the repo to much extra work for the user?
@dkrupp @bruntib what do you think?

Reply:
@gyorb In my above example I assumed that there will be always a root project directory which is a git repository (`root_project_dir`) but the parent directory of this directory is not. In this case my solution will not depend on the order of the compilation database. It will always get git information on the project root directory.

Is it possible that the compilation database will contain actions which do not have a common project directory which is a git repository?

So for example:
```
`- root_project_dir  # Not a git repository
     `- submodule1  # Git repository
       `- src1.cpp
      `- submodule2  # Git repository
       `- src2.cpp
```
And the compilation database contains 2 entries:
- /home/projects/root_project_dir/submodule1/src1.cpp
- /home/projects/root_project_dir/submodule2/src2.cpp

Reply:
If we can allow the extra cycles, we could use `git ls-files` and search for the particular file, and if and only if the Git thing shows the file, we can safely say it is part of that repository.